src/modules/functions.js

```diff
@@ -0,0 +1,153 @@
+  const parent = element.parentElement;
+  const ul = parent.children[2];
+  ul.classList.add('open-task-menu');
+  const newUl = document.querySelector('.open-task-menu');
+  window.addEventListener('mouseup', (e) => {
+    const isClose = e.target.closest('.open-task-menu');
+    if (!isClose && newUl.classList.contains('open-task-menu')) {
+      newUl.classList.remove('open-task-menu');
+    }
+  });
+};
+
+const showTasks = () => {
+  const parser = new DOMParser();
+  const ul = document.querySelector('.main-list');
+  ul.innerHTML = '';
+  let liStr = '';
+  allTasks.forEach((e, i) => {
+    if (e.completed === true) {
+      liStr = `<li>
+        <input type="checkbox" class="checkInput" id="input${i}" data-index ="${i}" checked><label for="input${i}">${e.description}</label>
+        <ul class="task-menu">
+            <li class="editTask" data-index ="${i}">
+                <span class="task-menu-text">Edit</span>
+                <span class="material-symbols-outlined">
+                    edit_square
+                </span>
+            </li>
+            <li class="deleteTask" data-index ="${i}">
+                <span class="task-menu-text">Delete</span>
+                <span class="material-symbols-outlined">
+                    delete
+                </span>
+            </li>
+        </ul>
+        <span class="material-symbols-outlined menu-btn">
+            more_vert
+        </span>
+        </li>`;
+    } else {
+      liStr = `<li>
+        <input type="checkbox" class="checkInput" id="input${i}" data-index ="${i}"><label for="input${i}">${e.description}</label>
+        <ul class="task-menu">
+            <li class="editTask" data-index ="${i}">
+                <span class="task-menu-text">Edit</span>
+                <span class="material-symbols-outlined">
+                    edit_square
+                </span>
+            </li>
+            <li class="deleteTask" data-index ="${i}">
+                <span class="task-menu-text">Delete</span>
+                <span class="material-symbols-outlined">
+                    delete
+                </span>
+            </li>
+        </ul>
+        <span class="material-symbols-outlined menu-btn">
+            more_vert
+        </span>
+        </li>`;
+    }
+    const li = parser.parseFromString(liStr, 'text/html').body.firstChild;
+    li.classList.add('task');
+    ul.append(li);
+  });
+  const editBtn = document.querySelectorAll('.editTask');
+  const updateBtn = document.querySelector('#edit-btn');
+  const addInput = document.querySelector('#add-input');
+  const menuBtn = document.querySelectorAll('.menu-btn');
+  const deleteBtn = document.querySelectorAll('.deleteTask');
+  const inputs = document.querySelectorAll('input');
+
+  inputs.forEach((e) => {
+    e.addEventListener('change', (ev) => {
+      ev.stopImmediatePropagation();
+      const { index } = ev.target.dataset;
+      setStatus(allTasks, index);
+      showTasks();
+    });
+  });
```

Comment:
I think this function is useless here because this function will be called only when we edit or add a new item and in those stages, we do not need to change the completed property on these stages it should be edited only when we click the checkbox button.
so to boost the performance you have to define a separate event listener just for your checkbox and run this function there.